src/lib/test-handler.js

```diff
@@ -115,13 +106,28 @@ export default class TestHandler {
           console.log(this.outputHandler.getSummaryOutput());
         }
       }
-    });
 
-    worker.on('error', (worker, error) => {
-      console.log('Error caught: ', error);
-    });
+      if (payload.exception) {
+        console.log('Error caught: ', payload.exception);
+      }
+    }.bind(this);
+
+    this.workers.push(worker);
 
-    return worker;
+    return worker.execute()
+    .then(function(result) {
+      return done(result);
+    })
+    .catch(function(err) {
+      return done({
```

Comment:
So sometimes while I'm testing something in the worker breaks and throws an exception separately from the reject :-/